src/components/form/model-provider/model-provider.spec.js

```diff
@@ -0,0 +1,236 @@
+        const input1 = inputs.at(0);
+        const input2 = inputs.at(1);
+
+        expect(wrapper.vm.value).to.be.equal('foo');
+        expect(input1.element.checked).to.be.ok;
+
+        input2.setChecked(true);
+
+        expect(wrapper.vm.value).to.be.equal('bar');
+        expect(input1.element.checked).to.be.not.ok;
+        expect(input2.element.checked).to.be.ok;
+    });
+
+    it('It supports input fields', () => {
+        const wrapper = shallowMount(Vue.extend({
+            components: runtimeComponents,
+            data() {
+                return {
+                    value: '',
+                };
+            },
+            template: `
+                <ui-model-provider v-model="value">
+                    <ui-input slot-scope="props" :value="props.value" @input="props.updateValue"/>
+                </ui-model-provider>
+            `,
+        }), {
+            stubs: runtimeComponents,
+        });
+
+        const input = wrapper.find('input');
+
+        expect(wrapper.vm.value).to.be.not.ok;
+        expect(input.element.value).to.be.not.ok;
+
+        input.setValue('foo');
+
+        expect(wrapper.vm.value).to.be.ok;
+        expect(input.element.value).to.be.ok;
+    });
+
+    it('It supports textarea fields', () => {
+        const wrapper = shallowMount(Vue.extend({
+            components: runtimeComponents,
+            data() {
+                return {
+                    value: '',
+                };
+            },
+            template: `
+                <ui-model-provider v-model="value">
+                    <ui-textarea slot-scope="props" :value="props.value" @input="props.updateValue"/>
+                </ui-model-provider>
+            `,
+        }), {
+            stubs: runtimeComponents,
+        });
+
+        const textarea = wrapper.find('textarea');
+
+        expect(wrapper.vm.value).to.be.not.ok;
+        expect(textarea.element.value).to.be.not.ok;
+
+        textarea.setValue('foo');
+
+        expect(wrapper.vm.value).to.be.ok;
+        expect(textarea.element.value).to.be.ok;
+    });
+
+    it('It supports multiple checkboxes', () => {
+        const wrapper = shallowMount(Vue.extend({
+            components: runtimeComponents,
+            data() {
+                return {
+                    value: [],
+                };
+            },
+            template: `
+                <ui-model-provider v-model="value">
+                    <div slot-scope="props">   
+                        <ui-checkbox :checked="props.hasItem('foo')" value='foo' @change="props.updateValue"/>
+                        <ui-checkbox :checked="props.hasItem('bar')" value='bar' @change="props.updateValue"/>
+                        <ui-checkbox :checked="props.hasItem('baz')" value='baz' @change="props.updateValue"/>
+                    </div>
+                </ui-model-provider>
+            `,
+        }), {
+            // checkboxes should have the sync option disabled
```

Comment:
Could you explain why in the comment?

Reply:
Fixed with https://github.com/dreipol/vue-ui/pull/9/commits/cbef53d2252eb777382aa48e6da556effad7c4c9